Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -84,12 +85,19 @@ open class ChatChannelVC: _ViewController,
         InvertedScrollViewPaginationHandler.make(scrollView: messageListVC.listView)
     }()
 
+    /// Determines if a messaged had been marked as unread in the current session
     private var hasMarkedMessageAsUnread: Bool {
-        channelController.firstUnreadMessageId != nil
+        channelController.markedAsUnread
     }
 
+    /// Unread messages count for the session.
+    /// It can either contain the initial unread count, or the one after marking the channel as read/unread
+    public var sessionUnreadCount: ChannelUnreadCount = .noUnread
```

Comment:
Why do we need to copy the unread count now? Why don't we use the `channelController.channel?.unreadCount` directly? Can't this be a bit error-prone?

Also, if the `sessionUnreadCount` lives in the UI Component, does it mean the Unread Feature is a UI feature? Meaing, that the SwiftUI SDK won't get this out of the box?

Reply:
This is in the ChatChannelVC because it is UI logic. We want to get the value for unread messages as we enter the chat and, independently of whether we mark the channel as read, keep showing the "unread messages" banner.

`channelController.channel?.unreadCount` will be updated to 0 unread as the channel is marked as read upon entering it